### PR TITLE
Wire players component to store

### PR DESCRIPTION
**Pull request by itslong**
-PlayersArea: acts as a Container to hold store state
-PlayersList: stateless and render.

expected behavior:
-initially, should display a message for user to enter a message in the Configuration.
-enters a number in the Configuration for players: ex: 5 (generates 5 users)
-a list of 5 users appear under Players (left column).
-clicking on the users removes them from the list and decrements the count. (these users will show up in the Queue, center column, in the future).
-when the list is empty, should display a message that there's no more users remaining.

1: initially display message and user has opened modal to input player number:

2: list of 6 users created and displayed:

3: 2 users were click and removed from the list:

### CODE REVIEW

File: src/actions/playersActions.js [hunk 1]
```diff
@@ -12,10 +18,23 @@ type Action =
   | UpdatePlayersListAction;
 
 
-function updatePlayersList(): Action {
+function updatePlayersList(id): Action {
```
Comment (ChrisW321) [inline src/actions/playersActions.js]: what is id for?

File: src/actions/playersActions.js [hunk 2]
```diff
@@ -12,10 +18,23 @@ type Action =
...
   return {
     type: UPDATE_PLAYERS_LIST
-  }
+  };
+}
+
+function initializePlayersList(total): Action {
+  const createdUsers = generateUsers(total);
+  const users = createdUsers.reduce((obj, currItem) => {
```
Comment (ChrisW321) [inline src/actions/playersActions.js]: obj as a variable name could be more descriptive, I would suggest for maps/objects to use the naming pattern valueByKey or keyToValue so in this case it looks like it would be usersByID or idToUsers

File: src/utility/generateUsers.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+
+function generateUsers(total, hasRating=false) {
+  /*
+  Return an array of user objects.
+  user: {
+    id: uuid
+    username: 
+  }
+  */
+  let users = [];
```
Comment (ChrisW321) [inline src/utility/generateUsers.js:10]: can be const

File: src/actions/playersActions.js (hunk 1 above)
Comment (itslong) [reply src/actions/playersActions.js]: When a user is clicked in the users list, the id is used to remove the user from the playersList state.

Shape of state is an object like this:

playersList: {
  1: {
    id: 1,
    username: 'user1'
  },
  2: {
    id: 2,
    username: 'user2'
  },
}

File: src/utility/generateUsers.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  while(total > 0) {
+    let user = {
+      id: total,
+      username: 'user-' + total.toString()
```
Comment (ChrisW321) [inline src/utility/generateUsers.js:14]: if you can we try to prefer string interpolation so `user-${total}`

File: src/utility/generateUsers.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    };
+    users.push(user);
+    total--;
+  }
+  return users;
+}
+
+export {
+  generateUsers
+};
```
Comment (ChrisW321) [inline src/utility/generateUsers.js:24]: could fit on one line

File: src/utility/generateUsers.js [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+  }
+  */
+  let users = [];
+  while(total > 0) {
+    let user = {
+      id: total,
+      username: 'user-' + total.toString()
+    };
+    users.push(user);
+    total--;
```
Comment (ChrisW321) [inline src/utility/generateUsers.js:17]: a fancy but less performant way to write this (I'm just code golfing)
return [...Array(total).keys()].reverse().map(num => ({id: num, username: `user-${num}`})

As a general guideline and not a strict rule: if a list is > 10-20 items performance trumps readability, if < than that readablity is better.

File: src/actions/playersActions.js [hunk 7]
```diff
@@ -20,7 +20,8 @@ type Action =
 
 function updatePlayersList(id): Action {
   return {
-    type: UPDATE_PLAYERS_LIST
+    type: UPDATE_PLAYERS_LIST,
+    playerId: id
```
Comment (ChrisW321) [inline src/actions/playersActions.js:33]: is see what id is for now

File: src/actions/playersActions.js (hunk 2 above)
Comment (itslong) [reply src/actions/playersActions.js]: Thanks, this is a great suggestion and I will adopt. I always have problems with naming these variables in map/objects.

File: src/components/PlayersArea.js [hunk 8]
```diff
@@ -1,11 +1,36 @@
...
+
+  const updatePlayer = (e) => {
+    const playerId = parseInt(e.target.id);
+    // NOTE: decrementing count and updating state is semi-redundant and could be combined into one call.
+    dispatch(playersCountDecrement());
+    dispatch(updatePlayersList(playerId));
+  };
+
   return (
     <div>
```
Comment (ChrisW321) [inline src/components/PlayersArea.js:25]: can this be a fragment?

File: src/components/PlayersArea.js [hunk 9]
```diff
@@ -1,11 +1,36 @@
...
-      Players
+      <p>Players: {playersCount}</p>
+      {playersCount > 0 && playersList ?
+        <PlayersList 
+          players={playersList}
+          handlePlayerOnClick={updatePlayer}
+        />
+        : totalPlayers > 0 ? 'No players remaining.'
```
Comment (ChrisW321) [inline src/components/PlayersArea.js:32]: nested ternaries is discouraged, this isn't too hard to follow tho

File: src/components/PlayersList.js [hunk 10]
```diff
@@ -0,0 +1,19 @@
+/* @flow */
+import React from 'react';
+
+
+function PlayersList({ players, handlePlayerOnClick }) {
+  return (
+      <ul onClick={handlePlayerOnClick} style={{ "listStyleType": "none", "display": "inline" }}>
+        {Object.entries(players).map(item => {
```
Comment (ChrisW321) [inline src/components/PlayersList.js]: since you're not using the key can prob just do this
Object.values(players).map({id: itemID, username} => (
            <li key={`${itemID + username}`} id={itemId}>{username}</li>
          )

File: src/components/PlayersArea.js (hunk 9 above)
Comment (itslong) [reply src/components/PlayersArea.js:32]: I didn't like this either. Would a better way to write this be to return early in the function body above the return, then a normal ternary in the return's body?

```
function PlayersArea() {
  const totalPlayers = useSelector(state => state.app.teamSettings.totalPlayers;
  ...
  // initial state is set to 0. Return earlier here
  if(totalPlayers == 0 && playersCount == 0) {
    return 'Use Configurations to set the number of players.'
  }

  return (
    <div>
    ...
    // keep the ternary here?
    {playersCount > 0 && playersList ?
      <PlayersList />
      : 'No players remaining.'
    }

    </div>
  )
}
```

File: src/components/PlayersList.js (hunk 10 above)
Comment (itslong) [reply src/components/PlayersList.js]: Great catch! I should have thought about it further and `Object.values` is much better.
